cluster-autoscaler/core/utils.go

```diff
@@ -293,6 +301,16 @@ func getNodeInfosForGroups(nodes []*apiv1.Node, cloudProvider cloudprovider.Clou
 			continue
 		}
 
+		// No good template, check cache of previously running nodes.
+		if exemplarNodes != nil {
+			if node, found := exemplarNodes[id]; found {
+				added, _ := processNode(node)
```

Comment:
I think it may work (i.e. that the logic is correct), but it seems to make a lot of assumptions about internal calls:
1) that processNode will not fail for a non-existing node just because it's gone - cache miss in cloud provider when retrieving NodeGroupForNode seems like the most obvious point of failure,
2) that BuildNodeInfo will work the same whether it exists or not - seems to be true for now, but it's not really part of any contract,
2) that seenGroups[id] will always be updated within processNode - not incorrect, just spreading logic.

Is there any reason we can't save sanitized node info and return it here when needed?

Reply:
Rewritten to cache nodeInfo